File: notification-service/src/main/java/dev/zvaryyka/notificationservice/models/NotificationType.java
package dev.zvaryyka.notificationservice.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;

@Getter
@Setter
@Entity
@Table(name = "notification_types")
public class NotificationType {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @ColumnDefault("nextval('notification_types_id_seq'::regclass)")
    @Column(name = "id", nullable = false)
    private Integer id;

    @Size(max = 50)
    @NotNull
    @Column(name = "type_name", nullable = false, length = 50)
    private String typeName;

}

File: rebalancer-service/src/main/kotlin/dev/zvaryyka/rebalancerservice/models/NotificationType.kt
package dev.zvaryyka.rebalancerservice.models

import jakarta.persistence.*
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter
import org.hibernate.annotations.ColumnDefault

@Entity
@Table(name = "notification_types")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
 class NotificationType {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @ColumnDefault("nextval('notification_types_id_seq'::regclass)")
    @Column(name = "id", nullable = false)
     var id: Int? = null

    @Size(max = 50)
    @NotNull
    @Column(name = "type_name", nullable = false, length = 50)
     var typeName: String? = null
}